src/Test/Html/Event.elm

```diff
@@ -1,307 +0,0 @@
-module Test.Html.Event
-    exposing
-        ( Event
-        , simulate
-        , expect
-        , toResult
-        , click
-        , doubleClick
-        , mouseDown
-        , mouseUp
-        , mouseEnter
-        , mouseLeave
-        , mouseOver
-        , mouseOut
-        , input
-        , check
-        , submit
-        , blur
-        , focus
-        )
-
-{-| This module lets you simulate events on `Html` values and expect that
-they result in certain `Msg` values being sent to `update`.
-
-
-## Simulating Events
-
-@docs Event, simulate, expect, toResult
-
-
-## Event Builders
-
-@docs click, doubleClick, mouseDown, mouseUp, mouseEnter, mouseLeave, mouseOver, mouseOut, input, check, submit, blur, focus
-
--}
-
-import Dict
-import ElmHtml.InternalTypes exposing (ElmHtml, ElmHtml(..), Tagger)
-import Json.Decode as Decode exposing (Decoder)
-import Json.Encode as Encode exposing (Value)
-import Test.Html.Query as Query
-import Test.Html.Query.Internal as QueryInternal
-import Expect exposing (Expectation)
-
-
-{-| A simulated event.
-
-See [`simulate`](#simulate).
-
--}
-type Event msg
-    = Event ( String, Value ) (QueryInternal.Single msg)
-
-
-{-| Simulate an event on a node.
-
-    import Test.Html.Event as Event
-
-    type Msg
-        = Change String
-
-
-    test "Input produces expected Msg" <|
-        \() ->
-            Html.input [ onInput Change ] [ ]
-                |> Query.fromHtml
-                |> Event.simulate (Event.input "cats")
-                |> Event.expect (Change "cats")
-
-You can simulate custom events by passing a tuple containing the event name string and a
-`Value` representing the event object the browser would send to the event listener callback.
-
-    import Test.Html.Event as Event
-    import Json.Encode as Encode exposing (Value)
-
-
-    type Msg
-        = Change String
-
-
-    test "Input produces expected Msg" <|
-        \() ->
-            let
-                simulatedEventObject : Value
-                simulatedEventObject =
-                    Encode.object
-                        [ ( "target"
-                          , Encode.object [ ( "value", Encode.string "cats" ) ]
-                          )
-                        ]
-            in
-                Html.input [ onInput Change ] [ ]
-                    |> Query.fromHtml
-                    |> Event.simulate ( "input", simulatedEventObject )
```

Comment:
I tried that, but it felt weird having a function whose implementation was `(,)`. Here's a PR adding it: https://github.com/eeue56/elm-html-test/pull/30

I don't have strong feelings about this, though. If you have a strong preference for `custom`, feel free to merge https://github.com/eeue56/elm-html-test/pull/30 !